src/schedulers.ts

```diff
@@ -42,11 +41,19 @@ export class PreemptiveScheduler implements Scheduler {
           }
         } catch (e) {
           if (/Maximum call stack/.test(e.toString())) {
-            const stacks: es.CallExpression[] = []
+            const stacks: any = []
             for (let i = 1; i <= 3; i++) {
-              stacks.push(context.runtime.frames[i - 1].callExpression!)
-            }
-            context.errors.push(new MaximumStackLimitExceeded(context.runtime.nodes[0], stacks))
+              let currentFrame = context.runtime.frames[i - 1]
+              while (!currentFrame.callExpression) {
```

Comment:
I still don't understand why the logic doesn't work. I assumed it work since it does give me the correct result when I ran it:

```javascript
Line 3: Infinite recursion
  i(1000, 1)..  i(1000, 1)..  i(1000, 1)..```

but when I changed it to

```javascript
function i(n, redundant) {
      return n === 0 ? 0 : 1 + i(n-1, r());
    }
    function r() {
      return 1;
    }
    i(1000, 1);
```
I do get ```
Line 3: Infinite recursion
  i(668, 1)..  i(667, 1)..  i(667, 1)..```.

which is wrong.

Reply:
Yeah that 668 667 667 is definitely wrong. The problem is that you're starting the search from frames[i], and frames[1] and frames[2] might both resolve to the same frame (in this case frames[2]). What you need to do is start the search from where you stopped last time, so you need to keep track of a j and pick the first 3 good frames.